stan/math/opencl/get_event_cl.hpp

```diff
@@ -0,0 +1,38 @@
+#ifndef STAN_MATH_OPENCL_GET_EVENT_CL_HPP
+#define STAN_MATH_OPENCL_GET_EVENT_CL_HPP
+#ifdef STAN_OPENCL
+
+#include <stan/math/opencl/matrix_cl.hpp>
+#include <stan/math/opencl/has_event_stack.hpp>
+#include <CL/cl.hpp>
+#include <type_traits>
+#include <vector>
+
+namespace stan {
+namespace math {
+namespace opencl_kernels {
+
+template <typename T, enable_if_no_event_stack<T> = 0>
```

Comment:
if we're filtering here in `get_event` (which could maybe be called `maybe_get_event` or something to say we're filtering) then we might not need to filter in `event_concat`? Which can then just be called `vec_concat` and templated on `T` instead of specific to `cl::Event`?

Reply:
Let me check but yeah I'm pretty sure that's true